lib/runner/job.ex

```diff
@@ -36,24 +36,17 @@ defmodule ElixirBench.Runner.Job do
   @doc """
   Executes a benchmarking job for a specific commit.
   """
-  def start_job(id, repo_slug, branch, commit, config) do
+  def start_job(job, opts) do
     ensure_no_other_jobs!()
 
-    job = %Job{
-      id: to_string(id),
-      repo_slug: repo_slug,
-      branch: branch,
-      commit: commit,
-      config: config
-    }
+    timeout = Keyword.get(opts, :timeout, Confex.fetch_env!(:runner, :job_timeout))
+    task_fun = Keyword.get(opts, :task_fun, &run_job/1)
```

Comment:
not sure if having the task_function be part of the options is the best solution here. Having it be part of the options sort of suggests to me that it's viable to pass something different in there. In reality we just want to have it for testing purposes.

I think using a default argument like `runner_function \\ &fun` works in our favor here

Reply:
Changed! :heavy_check_mark: